packages/runtime/src/widgets/DataGrid/DataGrid.tsx

```diff
@@ -88,12 +107,29 @@ export const DataGrid: React.FC<GridProps> = (props) => {
   return (
     <>
       <Table
+        dataSource={namedData}
+        key={resolvedWidgetId}
         onRow={(record, recordIndex) => {
           return { onClick: () => onTapActionCallback(record, recordIndex) };
         }}
-        dataSource={namedData}
-        key={resolvedWidgetId}
         ref={rootRef}
+        rowKey={(data: unknown) => {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+          const x = get(data, itemTemplate.name);
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
+          return x.key ?? "";
```

Comment:
@sibteali786 Can we make this as dynamic item, as each time there will be not a key, sometime user can use id or any other field also

Reply:
agree just on it.